codegen/src/CodeGen.scala

```diff
@@ -459,7 +466,7 @@ class CodeGen(implicit providerConfig: Config.ProviderConfig, typeMapper: TypeMa
 
     // the type has to match Pulumi's resource type schema, e.g. kubernetes:core/v1:Pod
     // please make sure, it contains 'index' instead of empty module part if needed
-    val typ = Lit.String(typeToken)
+    val typ = Lit.String(typeToken.sanitized.asString)
```

Comment:
Are we sure it's OK to modify the original unparsed token in any way?

Reply:
It is necessary, according to our own SDK validation, I'm not 100% sure the engine requires it. CC @lbialy

Reply:
I have an idea how to correct it. Coming soon.